src/components/AddResourceModal/AddResourceModalV2.js

```diff
@@ -43,59 +58,74 @@ const AddResourceModalV2 = props => {
-              top: '840px',
-              left: '20px',
-              boxShadow: 3
-            }}
-          >
-            <ChooseResource setFormStep={() => {}} />
-          </Box>
-        )
-      ) : (
-        props.open && (
-        <Paper
-          ref={wrapperRef}
-          open={props.open}
-          onClose={onClose}
-          sx={{
-            position: 'absolute',
-            left: '32px',
-            bottom: '133px'
-          }}
-        >
-          {fullScreen && (
-            <IconButton
-              aria-label="close"
-              onClick={onClose}
+      {fullScreen
+        ? props.toolbarModal == TOOLBAR_MODAL_RESOURCE && (
+            <Box
+              ref={refNode}
+              style={{
+                display:
+                  props.toolbarModal == TOOLBAR_MODAL_RESOURCE
+                    ? 'inline'
+                    : 'none'
+              }}
+              bgcolor={'white'}
               sx={{
+                borderRadius: '10px',
                 position: 'absolute',
-                right: 20,
-                top: 48,
-                color: theme => theme.palette.grey[500]
+                top: '840px',
+                left: '20px',
+                boxShadow: 3
               }}
-              size="large"
             >
-              <CloseIcon />
-            </IconButton>
-          )}
+              <ChooseResource setFormStep={() => {}} />
+            </Box>
+          )
+        : props.toolbarModal == TOOLBAR_MODAL_RESOURCE && (
+            <Paper
+              ref={wrapperRef}
+              onClose={onClose}
+              sx={{
+                position: 'absolute',
+                left: '32px',
+                bottom: '133px'
+              }}
+            >
+              {fullScreen && (
+                <IconButton
+                  aria-label="close"
+                  onClick={onClose}
+                  sx={{
+                    position: 'absolute',
+                    right: 20,
+                    top: 48,
+                    color: theme => theme.palette.grey[500]
+                  }}
+                  size="large"
+                >
+                  <CloseIcon />
+                </IconButton>
+              )}
 
-          <DialogContent>
-            <ChooseResource setFormStep={() => {}} />
-          </DialogContent>
-        </Paper>
-       )
-      )}
+              <DialogContent>
+                <ChooseResource setFormStep={() => {}} />
+              </DialogContent>
+            </Paper>
+          )}
     </>
   );
 };
 
-export default AddResourceModalV2;
+const mapStateToProps = state => ({
+  toolbarModal: state.toolbarModal
+});
+
+const mapDispatchToProps = {
+  TOOLBAR_MODAL_CONTRIBUTE,
+  TOOLBAR_MODAL_FILTER,
+  TOOLBAR_MODAL_RESOURCE,
+  TOOLBAR_MODAL_SEARCH,
+  TOOLBAR_MODAL_NONE,
+  setToolbarModal
+};
+
+export default connect(mapStateToProps, mapDispatchToProps)(AddResourceModalV2);
```

Comment:
This absolutely works but we should switch this over to the newer functional approach to using Redux state. We have a short guide on this in the redux_guide.md in this project and there's a live example in components/Head/Head.js